sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -874,13 +874,29 @@ class SessionCatalog(
     // the extra columns that we don't require), with UpCast (to make sure the type change is
     // safe) and Alias (to respect user-specified view column names) according to the view schema
     // in the catalog.
-    val projectList = viewColumnNames.zip(metadata.schema).map { case (name, field) =>
-      Alias(UpCast(UnresolvedAttribute.quoted(name), field.dataType), field.name)(
-        explicitMetadata = Some(field.metadata))
+    val projectList = if (isTempView) {
+      viewColumnNames.zip(metadata.schema).map { case (name, field) =>
+        Alias(UpCast(UnresolvedAttribute.quoted(name), field.dataType), field.name)(
+          explicitMetadata = Some(field.metadata))
+      }
+    } else {
+      viewColumnNames.zip(metadata.schema).
+        map { case (name, field) => innerStruct(Seq(), name, field)}
```

Comment:
nit format:
```
      viewColumnNames.zip(metadata.schema).map { case (name, field) =>
        innerStruct(Seq(), name, field)
      }
```

Reply:
You are right, i will make some test and fix also for array and map